### PR TITLE
Updated sam cli to read schemas region from boto3

### PR DESCRIPTION
**Pull request by gupta-n**
*Issue #, if available:*
EventBridge Schemas service regions are hardcoded in SAM CLI.

*Why is this change necessary?*
This change is necessary to avoid SAM CLI changes when schema service is available in new region.

*How does it address the issue?*
By reading schema regions from boto3.

*What side effects does this change have?*
Customer needs to use latest AWS SDK.

*Did you change a dependency in `requirements/base.txt`?*
    *If so, did you run `make update-reproducible-reqs`*
     Yes.

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [Done] Write unit tests
- [ Done] Write/update functional tests
- [ Done] Write/update integration tests
- [ Done] `make pr` passes
- [ NA] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samcli/lib/schemas/schemas_api_caller.py [hunk 1]
```diff
@@ -19,58 +19,82 @@ def list_registries(self, next_token=None, limit=10):
...
+                for registry in page["Registries"]:
+                    registries.append(registry["RegistryName"])
+            if not registries:
+                raise ResourceNotFound("No Registries found. This should not be possible, please raise an issue.")
+            next_token = page.get("NextToken", None)
+            return {"registries": registries, "next_token": next_token}
+        except EndpointConnectionError:
+            raise NotAvailableInRegion(
+                "EventBridge Schemas are not available in provided region. "
+                "Please check AWS doc for Schemas supported regions."
```
Comment (c2tarun) [inline samcli/lib/schemas/schemas_api_caller.py:38]: This string is repeated multiple times. Please move it to a constant.

Comment (c2tarun) [inline samcli/lib/schemas/schemas_api_caller.py:38]: This exact string is hardcoded at lot of places, please extract it to a constant.

File: requirements/base.txt [hunk 2]
```diff
@@ -1,7 +1,8 @@
 chevron~=0.12
 click~=7.0
 Flask~=1.0.2
-boto3~=1.10, >=1.10.29
+#Need to add Schemas latest SDK.
+boto3~=1.12, >=1.10.29
```
[the remark this answers is not in the document]
Comment (gupta-n) [reply requirements/base.txt]: sure.

File: samcli/lib/schemas/schemas_api_caller.py [hunk 3]
```diff
@@ -19,58 +19,82 @@ def list_registries(self, next_token=None, limit=10):
         if limit is None:
```
[the remark this answers is not in the document]
Comment (gupta-n) [reply samcli/lib/schemas/schemas_api_caller.py:37]: done

File: samcli/lib/schemas/schemas_api_caller.py (hunk 1 above)
Comment (gupta-n) [reply samcli/lib/schemas/schemas_api_caller.py]: done

File: requirements/base.txt [hunk 4]
```diff
@@ -1,7 +1,8 @@
 chevron~=0.12
 click~=7.0
 Flask~=1.0.2
-boto3~=1.10, >=1.10.29
+#Need to add Schemas latest SDK.
+boto3~=1.12, >=1.12.X
```
Comment (c2tarun) [inline requirements/base.txt:5]: Sorry for confusing comment, `x` was an example. Are you sure `1.12.X` works?

Comment (gupta-n) [reply requirements/base.txt:5]: I am waiting on new SDK will update.

**Pull request merged**